### ISSUE TITLE
incorporate the modular option to enable amd/umd support for generated views

### ISSUE THREAD

**Issue by simoami** [e-jigsaw/gulp-riot#37]
The riot command line has the following option which is missing in the gulp wrapper:

```
        if (opt.compiler.modular && i === 0) {
          out += '(function(tagger) {\n'
          out += '  if (typeof define === \'function\' && define.amd) {\n'
          out += '    define([\'riot\'], function(riot) { tagger(riot); });\n'
          out += '  } else if (typeof module !== \'undefined\' && typeof module.exports !== \'undefined\') {\n'
          out += '    tagger(require(\'riot\'));\n'
          out += '  } else {\n'
          out += '    tagger(window.riot);\n'
          out += '  }\n'
          out += '})(function(riot) {\n'
        }
```

It would be good to replicate this logic to reach out to the authors to internalize this command into the core compiler, as opposed to the cli logic.

**Comment by e-jigsaw** [comment]
@simoami Thanks for feature request. I will check code.

**Comment by e-jigsaw** [comment]
@simoami @chrisjob012
I released `v0.3.0` for this feature.

Thanks
